test/async_cache_test.dart

```diff
@@ -18,6 +18,14 @@ void main() {
     cache = AsyncCache(const Duration(hours: 1));
   });
 
+  test('should not fetch when callback throws exception', () async {
+    Future<String> asyncFunctionThatThrows() {
+      throw Exception();
+    }
+    var cacheFuture = cache.fetch(asyncFunctionThatThrows);
+    await expectLater(cacheFuture,throwsA(isException));
```

Comment:
This doesn't test that the cache is cleared on an error.

Consider:
```dart
var throwingFuture = cache.fetch(() async => throw StateError("Test"));
var secondThrowingFuture = cache.fetch(() async => "Unreachable");
// Future *is* cached until future completes.
expect(throwingFuture, same(secondThrowingFuture));

await expectLater(throwingFuture, throwsA(isStateError));

// Cache is cleared immediately after error future completes.
var valueFuture = cache.fetch(() async => "value");
await expectLater(valueFuture, "value");
```

Reply:
`expect(throwingFuture, same(secondThrowingFuture));`

What are we trying to check here ?

Reply:
Getting this error 

```
Expected: same instance as <Instance of 'Future<String>'>
Actual: <Instance of 'Future<String>'>
```

Reply:
I think we are getting this error because both functions are getting different memory address. And it is not able to distinguish.

Attaching supporting links [same method]("https://pub.dev/documentation/test/latest/test/same.html")